### PR TITLE
Better voting

### PR DESCRIPTION
**Pull request by VinaiRachakonda**

### CODE REVIEW

File: client/components/Voting/voting.tsx [hunk 1]
```diff
@@ -20,6 +21,8 @@ interface VotingState {
   finalvoteData: object;
   didFinalvote: boolean;
   votingFinalized: boolean;
+  prevotedPartners: Array<object>;
+  finalvotedPartners: Array<object>;
```
Comment (casey-chow) [inline client/components/Voting/voting.tsx]: These should be arrays of users.

File: client/components/Voting/voting.tsx [hunk 2]
```diff
@@ -34,7 +37,7 @@ const voteFormSchema = makeRequired(
   requiredFields
 );
 
-export default class VotingForms extends React.Component<
+export default class VotingForms extends React.PureComponent<
```
Comment (casey-chow) [inline client/components/Voting/voting.tsx]: You probably don’t want PureComponent here. It means react won’t re render until props change.

File: client/components/Voting/voting.tsx [hunk 3]
```diff
@@ -108,7 +117,39 @@ export default class VotingForms extends React.Component<
...
+      let newDidFinalvote = this.state.didFinalvote;
+      updatedCompany.partnerVotes.prevote.forEach((partnerObj: PartnerVoteObj) => {
+        newDidPrevote = newDidPrevote || (partnerObj.partner_id == this.props.user.id);
+      });
+
+      updatedCompany.partnerVotes.final.forEach((partnerObj: PartnerVoteObj) => {
+        newDidFinalvote = newDidFinalvote || (partnerObj.partner_id == this.props.user.id);
+      });
+
+      this.setState({prevotedPartners: updatedCompany.partnerVotes.prevote,
```
Comment (casey-chow) [inline client/components/Voting/voting.tsx]: It’s probably better to make a new timeout every time, that’ll kill a lot of redraw and timing errors.

File: client/components/Voting/voting.tsx [hunk 4]
```diff
@@ -233,50 +252,153 @@ export default class VotingForms extends React.Component<
...
-      alert('Missing votes');
-    }
-  }
-
   renderFinalizeVotesButton() {
     if (
       !this.state.didFinalvote ||
       this.props.user.partner_position !== 'Managing Partner'
     ) {
+      console.log('here');
```
Comment (casey-chow) [inline client/components/Voting/voting.tsx]: Remove

**Review by casey-chow — COMMENTED** [review]
Run `make fmt`

File: client/components/Voting/voting.tsx (hunk 2 above)
Comment (VinaiRachakonda) [reply client/components/Voting/voting.tsx]: oh whoops forgot  I changed that.

File: client/components/Voting/voting.tsx (hunk 1 above)
Comment (VinaiRachakonda) [reply client/components/Voting/voting.tsx]: Is it necessary to load the entire user? I defined a PartnerVoteObj instead.

File: client/components/Voting/voting.tsx (hunk 3 above)
Comment (VinaiRachakonda) [reply client/components/Voting/voting.tsx]: Sorry not quite sure I understand.

**Review by albertzhidong — APPROVED** [review]

**Pull request merged**